### PR TITLE
Make absence of pytest or numpy fatal at configure time

### LINKED ISSUE #849: Windows: short-tests are not run
> Prepare all required actions
> Run ./.github/workflows/run-tests
>   with:
>     simtest-tag: v8.3.0a
>     batch-name: short-tests
>     os: windows-2022
>   env:
>     VCPKG_ROOT: D:\a\Antares_Simulator\Antares_Simulator/vcpkg
>     pythonLocation: C:\hostedtoolcache\windows\Python\3.10.6\x64
>     ortools-tag: v9.2-rte1.3
> Run cd src/tests/resources/batches
>   cd src/tests/resources/batches
>   wget -q https://github.com/AntaresSimulatorTeam/SimTest/releases/download/v8.3.0a/short-tests-windows-2022.zip -O batch.zip
>   unzip -q batch.zip
>   rm batch.zip
>   shell: C:\Program Files\Git\bin\bash.EXE --noprofile --norc -e -o pipefail {0}
>   env:
>     VCPKG_ROOT: D:\a\Antares_Simulator\Antares_Simulator/vcpkg
>     pythonLocation: C:\hostedtoolcache\windows\Python\3.10.6\x64
>     ortools-tag: v9.2-rte1.3
> Run cd _build
>   cd _build
>   ctest -C Release --output-on-failure -R json
>   shell: C:\Program Files\Git\bin\bash.EXE --noprofile --norc -e -o pipefail {0}
>   env:
>     VCPKG_ROOT: D:\a\Antares_Simulator\Antares_Simulator/vcpkg
>     pythonLocation: C:\hostedtoolcache\windows\Python\3.10.6\x64
>     ortools-tag: v9.2-rte1.3
> Test project D:/a/Antares_Simulator/Antares_Simulator/_build
> **No tests were found!!!**
> Run rm -rf src/tests/resources/batches/*
>   rm -rf src/tests/resources/batches/*
>   shell: C:\Program Files\Git\bin\bash.EXE --noprofile --norc -e -o pipefail {0}
>   env:
>     VCPKG_ROOT: D:\a\Antares_Simulator\Antares_Simulator/vcpkg
>     pythonLocation: C:\hostedtoolcache\windows\Python\3.10.6\x64
>     ortools-tag: v9.2-rte1.3
>

### PR DESCRIPTION
**Pull request by flomnes**

### CODE REVIEW

File: .github/workflows/windows-vcpkg.yml [hunk 1]
```diff
@@ -92,10 +87,15 @@ jobs:
         antares-deps-version: ${{steps.antares-deps-version.outputs.prop}}
         os: ${{matrix.os}}
 
-    - name: Install dependencies
-      run: |
-          python -m pip install --upgrade pip
-          pip3 install -r src/tests/examples/requirements.txt
+    - name: Recover pip dependencies from cache
```
Comment (guilpier-code) [inline .github/workflows/windows-vcpkg.yml]: ok, why don't we do the same on linux ?

File: .github/workflows/windows-vcpkg.yml [hunk 2]
```diff
@@ -92,10 +87,15 @@ jobs:
...
+      uses: actions/setup-python@v4
```
Comment (guilpier-code) [inline .github/workflows/windows-vcpkg.yml:91]: If python is not found, will there be an error and will the CI stop ?

File: .github/workflows/windows-vcpkg.yml (hunk 1 above)
Comment (flomnes) [reply .github/workflows/windows-vcpkg.yml]: We could, yes

File: .github/workflows/windows-vcpkg.yml (hunk 2 above)
Comment (flomnes) [reply .github/workflows/windows-vcpkg.yml:91]: Yes

File: .github/workflows/windows-vcpkg.yml (hunk 1 above)
Comment (flomnes) [reply .github/workflows/windows-vcpkg.yml]: I'll revert that change since it's much longer than without cache (3min30s+30s instead of 30s)

**Pull request merged**